agent/php_user_instrument.c

```diff
@@ -507,6 +412,17 @@ nruserfn_t* nr_php_add_custom_tracer_named(const char* namestr,
   return wraprec; /* return the new wraprec */
 }
 
+#if ZEND_MODULE_API_NO >= ZEND_7_4_X_API_NO
+static void destroy_transient_wrapper(void* wraprec) {
+  nr_php_user_wraprec_destroy((nruserfn_t**)&wraprec);
+}
+
+void nr_php_init_transient_user_instrumentation(void) {
+  /* wraprecs are not re-usable and destroyed together with hashmap */
+  transient_wrappers = nr_hashmap_create(destroy_transient_wrapper);
```

Comment:
What should be the initial size of this hashmap? 64 for user_function_wrappers and 1024 for transient_wrappers?

Reply:
The only penalty for making it larger is the memory usage correct?  Each bucket is two pointers (for linked list) and a pointer to a wraprec?  If this is the case even 1024 seems completely reasonable in memory usage.

Reply:
Does making it larger make the lookups slower especially if it is sparsely populated?

Reply:
> Does making it larger make the lookups slower especially if it is sparsely populated?
I need to measure more closely how well the hashing function distributes the addresses of zend_functions but making the hashmap larger (more buckets) should reduce the lookup time because there will be less collisions to resolve (fewer keys, zend_function addresses, will be mapped to the same bucket).

Reply:
For now I made them 64 and 1024: 02406bda54989b60a5af873e30fe1d5740b676bb.

Reply:
It was actually 1e7ec63f5395d469264ad8c2ba5718f9546ba435 that made the hashmap size to be 1024. And hashing function distribution seems to work fairly ok. Some test results:
```
# elements: 1001, # buckets used: 649, collisions - min: 1, max: 29, avg: 2
```